### DISCUSSION TITLE
Supabase AI Assistant v2 feedback

### DISCUSSION THREAD

**Discussion by saltcod** [supabase/supabase#30855] [General]
We're excited to announce the release of our new Supabase Assistant v2 in the Dashboard.  Read all about it [here](https://supabase.com/blog/supabase-ai-assistant-v2).  

![thumb](https://github.com/user-attachments/assets/a4be9d55-5f5e-4f37-bb80-e289187550a7)

Access it from the top toolbar in the Dashboard
![screenshot-2024-12-03-at-14 32 21](https://github.com/user-attachments/assets/763e87ef-47e7-4345-8aa9-2ab9a6ca20c6)

What do you like most about it? Least?
Any features you find particularly useful?
Anything could be improved?
Anything missing?

Happy to hear your thoughts, suggestions, and experiences.

**Comment by saltcod** [comment]
From slack: 

A user asks for their functions and it says there are none but shows tables in public.   It can create and delete functions but does not know they exists.

**Reply by SaxonF** [reply]
PR that adds ability to get functions https://github.com/supabase/supabase/pull/31015

**Comment by tomaspozo** [comment]
Assistant gets stuck when asking it to fill table with sample data linked to a user (auth.users).

I have a table called customers, each customer belongs to a user. I want Assistant to create random customers for a given user. I used this prompt, and it got stock on that state (Thinking....):
![image](https://github.com/user-attachments/assets/d2aeaf3a-2e1f-4317-be11-5b9eebba649e)

If the user link is not requested (prompt: `create 10 random customer`), it works, but then the user_id key is empty on the SQL statement, and I have to fill it manually, or either adjust the prompt with `assign the customer to the following user_id 1234`

A more complex use case would have been: `create 10 random customers for each user`

**Comment by tomaspozo** [comment]
Not a big thing but this is what I get on local dev. It is weird to see a stranger avatar and being called Jhon Doe lol
<img width="365" alt="Screenshot 2024-12-03 at 10 02 12 AM" src="https://github.com/user-attachments/assets/bdf73e71-f4c8-4abd-9c06-2a2db6b7b825">

**Comment by tomaspozo** [comment]
Would be nice if the script included the statement for enabling RLS in the table. I did it manually.
<img width="484" alt="image" src="https://github.com/user-attachments/assets/5e9d7f8a-726d-489c-b4bf-02266401eb4a">

**Comment by tomaspozo** [comment]
Missing context after reseting. For example, I always asks for uuid for the id fields. When I keep prompting about table creations it respects that. But after a reset (I basically do it to clear out prompts) if I request a new table, I get bigints again.

Would be nice if somehow the assistant learns either from prompting or from current table structure this type of preferences.

**Comment by thelege2nd** [comment]
@saltcod two bugs, please release a fix for it.

first glitch. if you try to highlight long content to copy and scroll at the same time, it disappears. 

Bolded content is not copyable either. if you highlight and copy everything, only the non bolded content shows in the clipboard 

![image](https://github.com/user-attachments/assets/c6607dfe-2f1d-4199-9d27-6331702abf2a)

**Comment by tonnyavery** [comment]
Doesn't work on self-hosted supabase, fails with `http://localhost:8000/api/ai/sql/generate-v3` authentication failure even though the rest of the dashboard seems to work fine

**Comment by murphycj** [comment]
Some feedback. I asked the assistant if some feature was available in the Javascript SDK and it gave me wrong information (as far as I can tell). I was then able to convince it multiple times to change its mind. 🤦

<img width="914" alt="Screenshot 2025-05-16 at 3 37 58 PM" src="https://github.com/user-attachments/assets/f3c99d77-2bdd-4c5b-8db5-ffb6f13f7619" />
<img width="903" alt="Screenshot 2025-05-16 at 3 38 10 PM" src="https://github.com/user-attachments/assets/a09a529b-6faa-41b3-84a1-d5f342b82284" />
<img width="903" alt="Screenshot 2025-05-16 at 3 38 17 PM" src="https://github.com/user-attachments/assets/a160811d-e3ad-42fe-9686-9efb2be633e3" />
